operations/mimir-mixin/dashboards/ruler.libsonnet

```diff
@@ -94,20 +95,18 @@ local filename = 'mimir-ruler.json';
       $.row('Configuration API (gateway)')
       .addPanel(
         $.timeseriesPanel('QPS') +
-        $.qpsPanel('cortex_request_duration_seconds_count{%s, route=~"%s"}' % [$.jobMatcher($._config.job_names.gateway), ruler_config_api_routes_re])
+        $.qpsPanelNativeHistogram($.queries.ruler.requestsPerSecondMetric, utils.toPrometheusSelectorNaked($.jobSelector($._config.job_names.gateway) + [utils.selector.re('route', ruler_config_api_routes_re)]))
       )
       .addPanel(
         $.timeseriesPanel('Latency') +
-        $.latencyRecordingRulePanel('cortex_request_duration_seconds', $.jobSelector($._config.job_names.gateway) + [utils.selector.re('route', ruler_config_api_routes_re)])
+        $.latencyRecordingRulePanelNativeHistogram($.queries.gateway.requestsPerSecondMetric, $.jobSelector($._config.job_names.gateway) + [utils.selector.re('route', ruler_config_api_routes_re)])
       )
       .addPanel(
         local selectors = $.jobSelector($._config.job_names.gateway) + [utils.selector.re('route', ruler_config_api_routes_re)];
+        local labels = std.join('_', [matcher.label for matcher in selectors]);
+        local metricStr = '%(labels)s:%(metric)s' % { labels: labels, metric: $.queries.gateway.requestsPerSecondMetric };
         $.timeseriesPanel('Per route p99 latency') +
-        $.queryPanel(
-          'histogram_quantile(0.99, sum by (route, le) (%s:cortex_request_duration_seconds_bucket:sum_rate%s))' %
-          [$.recordingRulePrefix(selectors), utils.toPrometheusSelector(selectors)],
-          '{{ route }}'
-        ) +
+        $.perInstanceLatencyPanelNativeHistogram('0.99', metricStr, selectors, instanceLabel='route', from_recording=true) +
```

Comment:
Missing the legend :

Original:
![image](https://github.com/user-attachments/assets/e25a3190-1e73-4aed-8656-4194f26b3b74)

This PR (ignore dark mode):
![image](https://github.com/user-attachments/assets/a5a3f525-bb7e-4acd-89fd-1df09d3e3690)

Reply:
Thank you for pointing out this. I have fixed the missing legend.